logstash-core/src/main/java/org/logstash/Valuefier.java

```diff
@@ -19,22 +20,19 @@
 import org.logstash.bivalues.BiValues;
 import org.logstash.ext.JrubyTimestampExtLibrary;
 
-public class Valuefier {
+public final class Valuefier {
     private static final String PROXY_ERR_TEMPLATE = "Missing Valuefier handling for full class name=%s, simple name=%s, wrapped object=%s";
     private static final String ERR_TEMPLATE = "Missing Valuefier handling for full class name=%s, simple name=%s";
 
     private Valuefier(){}
 
-    private static Object convertJavaProxy(JavaProxy jp) {
-        if(jp == null) {
-            return BiValues.NULL_BI_VALUE;
-        }
-        Object obj = JavaUtil.unwrapJavaObject(jp);
+    private static Object convertJavaProxy(final JavaProxy jp) {
+        final Object obj = JavaUtil.unwrapJavaObject(jp);
         if (obj instanceof IRubyObject[]) {
             return ConvertedList.newFromRubyArray((IRubyObject[]) obj);
         }
         if (obj instanceof List) {
-            return ConvertedList.newFromList((List<Object>) obj);
+            return ConvertedList.newFromList((Collection<?>) obj);
```

Comment:
@jakelandis we only need a `Collection` here so no need to cast to `List` and hide possible simplifications from inspections. 
I mean I could also turn the check into `Collection`, but the problem I see with that is the behavior change. Then we could input a `Set` but get back a `List` from `ConvertedList`'s internal logic ... that wouldn't be a passive change but I guess we could add an issue to discuss relaxing this actually :)